ui/src/app/pages/data/data-set/dataset-page.tsx

```diff
@@ -519,60 +530,73 @@ const DataSetPage = fp.flow(withCurrentWorkspace(), withUrlParams())(
+          return {
+            domain: domain,
+            isLoading: true,
+            values: []
+          };
+        }),
+        previewDataLoading: true,
+        selectedPreviewDomain: domains[0]
+      });
+      domains.forEach(async domain => {
+        const request = {
+          domain: domain,
+          conceptSetIds: this.state.selectedConceptSetIds,
+          includesAllParticipants: this.state.includesAllParticipants,
+          cohortIds: this.state.selectedCohortIds,
+          prePackagedConceptSet: this.getPrePackagedConceptSet(),
+          values: this.state.selectedValues.map(domainValue => domainValue.value)
+        };
+        try {
+          const dataSetPreviewResp = await dataSetApi().generateDataSetPreview(namespace, id, request);
+          const domainPreviewIndex = '[' + fp.findIndex(['domain', domain], this.state.previewList) + ']';
+          this.setState({previewList: fp.set(domainPreviewIndex + '.isLoading', false,
+            fp.set(domainPreviewIndex + '.values', dataSetPreviewResp.values, this.state.previewList))});
+        } catch (ex) {
+          const exceptionResponse = await ex.json() as unknown as ErrorResponse;
+          let errorText: string;
+          switch (exceptionResponse.statusCode) {
+            case 400:
+              if (exceptionResponse.message ===
+                'Data Sets must include at least one cohort and concept.') {
+                errorText = exceptionResponse.message;
+              } else if (exceptionResponse.message ===
+                'Concept Sets must contain at least one concept') {
+                errorText = `One or more of your concept sets in domain ${domain}
+                  has no concepts. Please check your concept sets to ensure all
+                  concept sets have concepts.`;
+              }
+              break;
+            case 404:
+              if (exceptionResponse.message.startsWith(
+                'Not Found: No Cohort definition matching cohortId')) {
+                errorText = 'Error with one or more cohorts in the data set. ' +
+                  'Please submit a bug using the contact support button';
+              }
+              break;
+            case 504:
+              if (exceptionResponse.message ===
+                'Timeout while querying the CDR to pull preview information.') {
+                errorText = `Query to load data from the All of Us Database timed out for domain:
+                  ${domain}. Please either try again or export data set to a notebook to try
+                  there`;
+              }
+              break;
+            default:
+              errorText = `An unexpected error has occurred loading domain: ${domain}.
+              Please submit a bug using the contact support button`;
+          }
+          this.setState({previewError: true, previewErrorText: errorText});
+          console.error(ex);
+        } finally {
+          this.setState({previewDataLoading: false});
```

Comment:
Yes, but on thinking more about this, we may not even need this piece of state anymore, because it is derivable from the `isLoading` bool on each domain.